BUG : assignment of non-ns timedelta64 value (GH14155)

rsmith54 opened the pull request:
- [x] closes #14155 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

jreback commented on pandas/core/internals.py:
```diff
@@ -1696,7 +1696,7 @@ def _try_coerce_args(self, values, other):
         elif isinstance(other, np.timedelta64):
             other_mask = isnull(other)
-            other = other.view('i8')
+            other = Timedelta(np.timedelta64(other))
```
use Timedelta(other).value

jorisvandenbossche replied:
Can you address this one? 
The `np.timedelta64(..)` is not needed given you just tested that it is already a `np.timedelta64` object

jreback commented:
tests might be in tseries/test_timedelta.py but more likely in index/test_indexing.py

Commit:
BUG : fix for GH14155

rsmith54 commented:
Added the call to other, and added a test in ```tests/series/test_indexing.py```.  Please let me know if there is any other comments or changes needed.

jorisvandenbossche commented on doc/source/whatsnew/v0.19.0.txt:
````diff
@@ -1478,3 +1478,4 @@ Bug Fixes
 
 - Bug in ``eval()`` where the ``resolvers`` argument would not accept a list (:issue:`14095`)
 - Bugs in ``stack``, ``get_dummies``, ``make_axis_dummies`` which don't preserve categorical dtypes in (multi)indexes (:issue:`13854`)
+- Bug in ``internals._try_coerce_args``, when setting ```Timedelta``` with ```ix```  (:issue:`14155`)
````
Can you make this clearer for the user? (specify the user-visible effect of the bug, not the underlying reason)

jorisvandenbossche commented on pandas/tests/series/test_indexing.py:
```diff
@@ -1324,6 +1324,13 @@ def test_timedelta_assignment(self):
 
+        # GH 14155
+        s = Series(10*[np.timedelta64(10, 'm')])
+        s.ix[[1,2,3]] = np.timedelta64(20, 'm')
```
Can you use `loc` here instead of `ix`? (it has the same issue, and is preferred to use over `ix`)

jorisvandenbossche commented on pandas/tests/series/test_indexing.py:
```diff
@@ -1324,6 +1324,13 @@ def test_timedelta_assignment(self):
+        s.ix[[1,2,3]] = np.timedelta64(20, 'm')
+        expected = pd.Series(10*[np.timedelta64(10, 'm')])
+        expected.ix[[1,2,3]] = pd.Timedelta(np.timedelta64(20, 'm'))
+        tm.assert_series_equal(s, expected)
```
You will also have to clean those lines above for PEP8 issues (eg whitespace around operators or after comma's). Travis will also complain (by a failing lint test) about this.

rsmith54 commented:
I have edited the ```whatsnew``` to be a bit clearer, changed ```ix``` to ```loc```, and fixed the problems with PEP8.

jreback commented on pandas/core/internals.py:
```diff
@@ -1696,7 +1696,7 @@ def _try_coerce_args(self, values, other):
         elif isinstance(other, np.timedelta64):
             other_mask = isnull(other)
-            other = other.view('i8')
+            other = Timedelta(np.timedelta64(other)).value
```
``Timesdelta(other).value`` is idiomatic

rsmith54 replied:
Oops, yep of course.  Fixed.

jreback commented:
minor comments. lgtm. ping on green.

rsmith54 commented:
@jreback , looks like all checks have passed.  Let me know if there is anything else I can do.

jorisvandenbossche commented:
@rsmith54 Maybe something went wrong with committing the changes, but it seems the `` Timedelta(np.timedelta64(other))`` has not changed

jreback commented:
lgtm.

jreback commented:
thanks!
